src/hybrid_reader.c

```diff
@@ -0,0 +1,329 @@
+#include "hybrid_reader.h"
+#include "util/heap.h"
+#include "VecSim/vec_sim.h"
+#include "VecSim/query_results.h"
+
+typedef enum {
+  STANDARD_KNN,
+  HYBRID_ADHOC_BF,
+  HYBRID_BATCHES
+} VecSearchMode;
+
+typedef struct {
+  IndexIterator base;
+  VecSimIndex *index;
+  TopKVectorQuery query;
+  VecSimQueryParams runtimeParams;   // evaluated runtime params.
+  IndexIterator *childIt;
+  VecSearchMode mode;
+  bool RESULTS_PREPARED;
+  VecSimQueryResult_List list;
+  VecSimQueryResult_Iterator *iter;
+  t_docId lastDocId;
+  RSIndexResult **returnedResults; // Save the pointers to be freed in clean-up.
+  char *scoreField;  // to use by the sorter, for distinguishing between different vector fields.
```

Comment:
not used

Reply:
This is on purpose (as I explained in previous comment)